### PR TITLE
Add Container Deployments from GitHub Actions to deploy-from-github.md

### PR DESCRIPTION
**Pull request by jtrugman**
Added instructions about how to build containers in github actions and deploy them to CapRover using the GitHub container registry

### CODE REVIEW

**Comment by githubsaturn** [comment]
Thanks for the PR!

This is marked as a draft - assuming you're still working on it. Let me know once it's ready to review.

**Comment by jtrugman** [comment]
No problem! I appreciate the community's help with me deploying pre-built images using github actions! 

I summarized the steps of what worked in the docs so future caprover users can learn how to do it from there. Please let me know your feedbacks.

File: docs/ci-cd-integration/deploy-from-github.md [hunk 1]
```diff
@@ -102,7 +102,98 @@ When you commit files to your project's repo on the "main" branch, Github Action
 
 ### Alternative method (more efficient)
 
-Alternatively, you can even build the Docker image on Github and just deploy the built artifact to your CapRover instance. This will help as it does not consume RAM and CPU from your CapRover instance to build your image. See this [comment](https://github.com/caprover/caprover/issues/1514#issuecomment-1250369318) for more details.
+Alternatively, you can even build the Docker image on Github and just deploy the built artifact to your CapRover instance. This will help as it does not consume RAM and CPU from your CapRover instance to build your image.
+
+In order to achieve this we will need to take the following steps to build the Docker image using GitHub Actions, store it using GitHub Packages, and then deploy it to CapRover.
+
+#### Create a GitHub Personal Access Token
+You will need to create a GitHub Personal Access Token with write permission for packages.
```
Comment (githubsaturn) [inline docs/ci-cd-integration/deploy-from-github.md:110]:
```suggestion
You will need to create a GitHub Personal Access Token with **write permission for packages**.
```

File: docs/ci-cd-integration/deploy-from-github.md [hunk 2]
```diff
@@ -102,7 +102,98 @@ When you commit files to your project's repo on the "main" branch, Github Action
...
+
+GitHub has a great guide on creating a personal access token if you have not before. Here is the link: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
+
+#### Create a New App
+If you do not have an app already on CapRover, create one using the instructions [here](### Create a new App)
```
Comment (githubsaturn) [inline docs/ci-cd-integration/deploy-from-github.md:115]: `### Create a new App` where do these links lead to? I don't see a linked page here. Same with the other ones below:
- Enable App Token
- Add the Github Secrets
- Add files to project

File: docs/ci-cd-integration/deploy-from-github.md [hunk 3]
```diff
@@ -102,7 +102,98 @@ When you commit files to your project's repo on the "main" branch, Github Action
...
+      
+    - name: Login to Container Registry
+      uses: docker/login-action@v2
+      with:
+            registry: ghcr.io
+            username: ${{ github.repository_owner }}
+            password: ${{ secrets.GH_TOKEN }}
+
+    - name: Preset Image Name
+      run: echo "IMAGE_URL=$(echo ghcr.io/${{ github.repository_owner }}/${{ github.event.repository.name }}:$(echo ${{ github.sha }} | cut -c1-7) | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
```
Comment (githubsaturn) [inline docs/ci-cd-integration/deploy-from-github.md:165]: Is this line well tested? I think previously we had this:
```
            - run: |
                  echo "IMAGE_NAME_WITH_REGISTRY=$DOCKER_REGISTRY/$IMAGE_NAME" >> $GITHUB_ENV
                  export IMAGE_NAME_WITH_REGISTRY=$DOCKER_REGISTRY/$IMAGE_NAME
                  echo "FULL_IMAGE_NAME=$IMAGE_NAME_WITH_REGISTRY:$GITHUB_SHA-gitsha" >> $GITHUB_ENV
```

If this line is well tested, I am okay with using this new line as it's simpler.

**Review by githubsaturn — COMMENTED** [review]
Thanks!

Just a few minor comments / questions.

File: docs/ci-cd-integration/deploy-from-github.md (hunk 2 above)
Comment (jtrugman) [reply docs/ci-cd-integration/deploy-from-github.md:115]: They link to the section above in the page. In this specific example it links to the ### Create a new App section on line 11.

File: docs/ci-cd-integration/deploy-from-github.md (hunk 3 above)
Comment (jtrugman) [reply docs/ci-cd-integration/deploy-from-github.md:165]: I borrowed this line from @maietta's example: https://github.com/maietta/SvelteKit-Prisma-Foundation-Boilerplate/blob/master/.github/workflows/production.yml

It was worked well for me so far.

Open to changing it if you want to have your suggested commands instead.

Comment (maietta) [reply docs/ci-cd-integration/deploy-from-github.md:165]: Hi @jtrugman. Glad you found my one-liner useful. The reason I had to use craft this version was to address malformed docker image names as docker images I think require lowercase tagging. By going all lowercase, it guarantied there were no issues with creating and pushing the image to Github Container Registry.

**Comment by githubsaturn** [comment]
Thank you!!

**Pull request merged**